### PR TITLE
Add 3.11-dev pipeline

### PR DESCRIPTION
**Pull request by Kludex**
Related: https://github.com/encode/uvicorn/pull/1304

### CODE REVIEW

**Comment by Kludex** [comment]
This depends on https://github.com/MagicStack/uvloop/issues/450

**Comment by Kludex** [comment]
In two weeks I'll come here to bother again - 3.11 beta will be up. 🏃 💨

**Comment by elprans** [comment]
The blocker here is upstream Cython release with proper support for 3.11.

**Comment by Kludex** [comment]
The PR you mentioned on https://github.com/MagicStack/uvloop/issues/450#issuecomment-964584584 was already released there 🤔

**Comment by elprans** [comment]
There have been _more_ breakages since then, fixes for which, AFAIK, haven't been released yet.  Generally speaking, until the ABI is declared stable (which is usually rc1), it's not worth trying to chase the next Python version because Cython is very sensitive to any upstream changes due to how heavily it uses the C API.

**Comment by Kludex** [comment]
Ah ok. Thanks! :)

I'll wait then. 🙏

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9, 3.10.0-rc.1]
+        python-version: [3.7, 3.8, 3.9, 3.10.0-rc.1, "3.11-dev"]
```
Comment (cclauss) [inline .github/workflows/tests.yml:17]:
```suggestion
        python-version: [3.7, 3.8, 3.9, "3.10", "3.11-dev"]
```
https://dev.to/hugovk/the-python-3-1-problem-85g

**Comment by cclauss** [comment]
Please rebase to remove conflicts.

**Comment by Kludex** [comment]
I'm not jealous, the other PR can take the lead. 🙏

**Pull request closed without merging**